### DISCUSSION TITLE
Symfony mailer : all emails are not sent if one of the recipients have an unknown domain

### DISCUSSION THREAD

**Discussion by sdespont** [symfony/symfony#50495] [Q&A]
Hello,

I need your help to understand why an error is occurring aborting the email sending when sending email with an unknown domain in the recipients list. The expected behaviour would be not stopping the email sending process is one of the emails has an unknown domain (all the valid emails must received the email). It was the behaviour of SwiftMailer which I was using before.

For example, I use the recipient "anicetest@gmail.com" which has a valid domain, and "test@test.com222" which has an unknown domain. When sending the email with Symfony Mailer, a `TransportException` is throwing and "anicetest@gmail.com" never received the email. Why?

Here is the error debug text

```
   < 220 mail.infomaniak.com ESMTP ready
    > EHLO [127.0.0.1]
    < 250-mail.infomaniak.com
    < 250-PIPELINING
    < 250-SIZE
    < 250-ETRN
    < 250-ENHANCEDSTATUSCODES
    < 250-8BITMIME
    < 250-DSN
    < 250 AUTH PLAIN LOGIN
    > AUTH LOGIN
    < 334 VXNlcm5hbWU6
    > YWRtaW4taW5mb0BtZWFpbW1vLmNo
    < 334 UGFzc3dvcmQ6
    > Y2EwMDkxY2YtNzEyMC00MTQ4LTlhMjQtOTY4OTFhNDExNkMy
    < 235 2.0.0 OK
    > MAIL FROM:<contact@test.com>
    < 250 2.1.0 Ok
    > RCPT TO:<anicetest@gmail.com>
    < 250 2.1.5 Ok
    > RCPT TO:<test@test.com222>
    < 450 4.1.2 <test@test.com222>: Recipient address rejected: Domain not found
```

![image](https://github.com/symfony/symfony/assets/646632/b4a98598-597f-4f06-9aa9-0ba74061e0f3)

**Comment by elevado** [comment]
Google's mail servers are trying to resolve the domain "test.com222" with DNS-Lockup. This is not even a real domain in format.

_450 4.1.2 [test@test.com222](mailto:test@test.com222): Recipient address rejected: **Domain not found**_

This is done to block spam. Therefore, use a real domain in the sender email.

**Reply by sdespont** [reply]
Thank you for your answer. If I use a recipient email like "test@testhdjkhsdfh.com" (this domain does not exist) the same error occurs and none of the emails are sent. Do I have to check if the email's domain exists for all the emails addresses before sending? It seems wrong and all the emails with a valid domain should be sent, isn't it?

**Reply by elevado** [reply]
It can be assumed that Google's mail servers do a DNS lookup (MX), which is pretty standard these days. Often they will even check that the email was sent from the email server that the MX record points to.

You can, of course, do a DNS lookup yourself:
* https://www.php.net/manual/en/function.checkdnsrr.php
* https://github.com/spatie/dns

I have never made a DNS lookup in applications myself. You'll have to talk to your provider to check if they allow you to do it or if there is a public DNS server you can use for free or paid.

In your case, can users send email to other users through an application? You have to be careful with that. You will end up on blacklists more quickly if you use it to generate spam.

If the emails are only going to you, I would ignore them. But you also have to protect your application so that it can't be misused.
